web-animations-2/Overview.bs

```diff
@@ -1066,13 +1066,36 @@ condition:
 > *   the <a>animation effect</a> has a <a>parent group</a> and the
 >     <a>parent group</a> is <a>current</a>.
 
-Add the following definition to the list used in determining the phase of
+Add the following definitions to the list used in determining the phase of
 an <a>animation effect</a>.
 
 :   <dfn>effective timeline phase</dfn>
 ::  The associated <a>animation</a>'s <a>hold phase</a> if resolved, otherwise
     the <a>animation</a> timeline's [=timeline phase=].
 
+:   <dfn>effective timeline progress</dfn>
+::  Set based on the first matching condition:
+
+    <dl class="switch">
+
+    :   If <em>all</em> of the following conditions are true:
+        *   the associated <a>animation</a>'s [=timeline duration=] is resolved
+            and non-zero, and
+        *   the associated <a>animation</a>'s [=playback rate=] is 1, and
```

Comment:
Was previously trying to avoid some edge cases for paused animations, but in the process, it needlessly restricted check on running animations.  in this version, we are assuming that the timeline and animation are aligned when paused, but otherwise adjust for changes to the playback rate and/or start time.  Not sure of a better option unless storing off a "hold timeline time" instead of the "hold phase".  Having said that, getting the phase at a boundary wrong for a scroll after pause animation is likely not that critical.  Authors can always specify a fill to get the expected behavior if they expect that the scroll-linked animation will be intermittently paused.

In short, generalized "effective timeline progress" and added "at timeline boundary".  The boundary does not discriminate based on direction since the boundary's counterpart is already inclusive.  We are making both endpoints inclusive if "at timeline boundary" is set.